docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+}
+
+ACTION hello::goodbye(name nm) {
+
+   balance_table balances(get_self(), get_self().value);
+
+   // check if the user already exists
+   auto itr = balances.find(nm.value);
+   if ( itr != balances.end() ) {
+      balances.erase(itr);
+   }
+}
+```
+
+### Multi-index: Test
+
+After you build and deploy again the smart contract you can send the `hi` action a couple of times and observe the results:
+
+```shell
+dune --send-action hello hi '[ama]' hello@active
+```
+
+```txt
+executed transaction: a92e307fa78f20a10f8639453051940f8f50b2db94e20a2ccd7c5304def6232e  104 bytes  415 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::hi                    {"nm":"ama"}
+>> Name: ama, balance: 1.0000 EOS
+#         hello <= hello::gettime               ""
+>> Year: 2023, Month: 2, Day: 24, Hour: 14, Minute: 37, Second: 2
+```
+
+```shell
+dune --send-action hello hi '[ama]' hello@active
+```
+
+```txt
+executed transaction: a92e307fa78f20a10f8639453051940f8f50b2db94e20a2ccd7c5304def6232e  104 bytes  415 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::hi                    {"nm":"ama"}
+>> Name: ama, balance: 0.9999 EOS
+#         hello <= hello::gettime               ""
+>> Year: 2023, Month: 2, Day: 24, Hour: 14, Minute: 37, Second: 2
+```
+
+```shell
+dune --send-action hello goodbye '[ama]' hello@active
+```
+
+Note in the outputs above that the balance started with 1 EOS and then it decreased to 0.9999 EOS.
+
+```txt
+executed transaction: 68ceadffa8858e094cb1bce742d0eccc6e8f7eece5e5e650850845183f8109cc  104 bytes  249 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::goodbye               {"nm":"ama"}
+```
+
+```shell
+dune --send-action hello hi '[ama]' hello@active
+```
+
+```txt
+executed transaction: a92e307fa78f20a10f8639453051940f8f50b2db94e20a2ccd7c5304def6232e  104 bytes  415 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::hi                    {"nm":"ama"}
+>> Name: ama, balance: 1.0000 EOS
+#         hello <= hello::gettime               ""
+>> Year: 2023, Month: 2, Day: 24, Hour: 14, Minute: 37, Second: 2
+```
+
+Note in the outputs above that after the `goodbye` action is successfully executed the `hi` action re-initializes the `ama` account balance to 1 EOS.
+
+## Singleton
+
+A singleton is a special type of multi-index table that is designed to store a single row of data. Unlike a regular multi-index table, which can store any number of rows, a singleton is restricted to a single row. Singletons are often used to store global state variables or configuration parameters in a contract. Because there is only one row in a singleton, it is more efficient to access and modify its data than a regular multi-index table.
```

Comment:
I don't think it's actually more efficient. It's still a multi-index underneath, though there might be some RAM savings for lack of indexes. Let's just omit that part now though. 

Definition also sounds like it can't be used for more than one thing stored, which isn't true. Scope can allow you to save 1 item per scope which makes it good for per-account configs (as an example) as well.

Reply:
per account configs is a good point.